acceptance/openstack/containerinfra/v1/containerinfra.go

```diff
@@ -0,0 +1,92 @@
+package v1
+
+import (
+	"testing"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/acceptance/clients"
+	"github.com/gophercloud/gophercloud/acceptance/tools"
+	"github.com/gophercloud/gophercloud/openstack/containerinfra/v1/clustertemplates"
+	th "github.com/gophercloud/gophercloud/testhelper"
+)
+
+// CreateClusterTemplate will create a random cluster tempalte. An error will be returned if the
+// cluster-template could not be created.
+func CreateClusterTemplate(t *testing.T, client *gophercloud.ServiceClient) (*clustertemplates.ClusterTemplate, error) {
+	choices, err := clients.AcceptanceTestChoicesFromEnv()
+	if err != nil {
+		return nil, err
+	}
+
+	name := tools.RandomString("TESTACC-", 8)
+	t.Logf("Attempting to create cluster template: %s", name)
+
+	createOpts := clustertemplates.CreateOpts{
+		Name:                name,
+		Labels:              map[string]string{},
+		FixedSubnet:         "",
+		MasterFlavorID:      "",
+		NoProxy:             "10.0.0.0/8,172.0.0.0/8,192.0.0.0/8,localhost",
+		HTTPSProxy:          "http://10.164.177.169:8080",
+		TLSDisabled:         false,
+		KeyPairID:           "kp",
```

Comment:
Will this fail if a keypair of "kp" doesn't exist? You can import `github.com/gophercloud/gophercloud/acceptance/openstack/compute/v2` and use https://github.com/gophercloud/gophercloud/blob/master/acceptance/openstack/compute/v2/compute.go#L275 to generate a keypair:

```go
import (
  compute "github.com/gophercloud/gophercloud/acceptance/openstack/compute/v2"
)

func CreateClusterTemplate(...) (...) {
   keypair, err := compute.CreateKeyPair(t, computeClient)
}
```

Reply:
Yes...this one will probably be defined from env var.

Reply:
The other option is to not specify a keypair at all. Since I don't think any test is going to actually SSH to the created cluster, it should be safe to omit.